beetsplug/replaygain.py

```diff
@@ -312,10 +334,36 @@ def end_element_handler(name):
                                           'the output of bs1770gain')
                 album_gain["album"] = Gain(state['gain'], state['peak'])
                 state['gain'] = state['peak'] = None
+            elif len(per_file_gain) == len(path_list):
+                if state['gain'] is not None:
+                    album_state['gain'] = state['gain']
+                if state['peak'] is not None:
+                    album_state['peak'] = state['peak']
+                if album_state['gain'] is not None \
+                        and album_state['peak'] is not None:
+                    album_gain["album"] = Gain(
+                        album_state['gain'], album_state['peak'])
+                state['gain'] = state['peak'] = None
+
         parser.StartElementHandler = start_element_handler
         parser.EndElementHandler = end_element_handler
 
         try:
+            # Sometimes, the XML out put of `bs1770gain` gets spliced with
+            #   some progress percentages: b'9%\x08\x0810%\x08\x08'
+            #   that are supposed to be canceled out by appending
+            #   a b'\x08' backspace characters for every character,
+            #
+            # For some reason, these backspace characters don't get
+            #   resolved, resulting in mangled XML.
```

Comment:
Aha; that's pretty weird! And there's no chance there's a "silent" flag we can pass to the tool to suppress the progress output?

Reply:
There is, I had overlooked it apparently!